app/src/org/commcare/fragments/SelectInstallModeFragment.java

```diff
@@ -74,9 +75,8 @@ public void onClick(View v) {
                     if (currentActivity instanceof CommCareSetupActivity) {
                         ((CommCareSetupActivity)currentActivity).clearErrorMessage();
                     }
-                    Intent i = new Intent("com.google.zxing.client.android.SCAN");
-                    i.putExtra("SCAN_FORMATS", "QR_CODE");
-                    getActivity().startActivityForResult(i, CommCareSetupActivity.BARCODE_CAPTURE);
+                    Intent intent = new IntentIntegrator(getActivity()).createScanIntent();
```

Comment:
Are there any potential failure points of this call? Like only working above a certain Android API level or something? It seems worth testing this on a few different devices if you haven't already (or at least flagging to the QA team that they should be sure to do so themselves)